src/test-api.c

```diff
@@ -117,6 +129,9 @@ static void test_api_generic(void) {
                 for (i = 0; i < sizeof(fns) / sizeof(*fns); ++i)
                         c_assert(!!fns[i]);
         }
+
+        /* test c_assert(0) does not result in -Wsometimes-uninitialized warnings. */
+        c_assert(_check_cassert_unreachable(1) == 1);
```

Comment:
We use `test-api.c` across all c-util projects to test for symbol and API availability. Unlike the other tests, it is always dynamically linked and can thus test that symbol-files work as expected, and that we do not accidentally drop symbols in future releases. It is not meant to test for runtime correctness (yet some old tests there do, I know).

Generally, we want basic tests like this in `test-basic.c`, or separate test-cases.